webhook/certs.go

```diff
@@ -47,7 +47,12 @@ func createCertTemplate(name, namespace string) (*x509.Certificate, error) {
 	}
 
 	serviceName := name + "." + namespace
-	serviceNames := []string{serviceName, serviceName + ".svc", serviceName + ".svc.cluster.local"}
+	serviceNames := []string{
```

Comment:
What's the reasoning for allowing non-namespaced address here?

Reply:
that's the point, we should add the svc name to DNSNames field.

as the kubernetes service docs said [what-things-get-dns-names](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#what-things-get-dns-names)

```
Assume a Service named **foo** in the Kubernetes namespace **bar**. A Pod running in namespace bar can look up this service by simply doing a DNS query for **foo**. A Pod running in namespace quux can look up this service by doing a DNS query for foo.bar.
```

we can directly use svc name (foo) to access current namespace's svc (foo.bar)

WDYT